cpp/oneapi/dal/algo/finiteness_checker/compute_types.cpp

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/finiteness_checker/compute_types.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::finiteness_checker {
+
+template <typename Task>
+class detail::v1::compute_input_impl : public base {
+public:
+    compute_input_impl(const table& x) : x(x) {}
```

Comment:
it makes sense to refactor this compute input, and rename it. as example in dbscan:
`namespace oneapi::dal::dbscan {

template <typename Task>
class detail::v1::compute_input_impl : public base {
public:
    compute_input_impl(const table& data) : data(data) {}
    compute_input_impl(const table& data, const table& weights) : data(data), weights(weights) {}

    table data;
    table weights;
};`